feature-selection/incremental.metta

```diff
@@ -0,0 +1,254 @@
+   (if (<= (List.length $indices) 1)
+       Nil  ;; Can't have redundancy with 1 or fewer features
+       (findRedundantSubsets $itable $indices $redundancyThreshold 1)))
+
+
+
+
+;; Process interaction terms of each size iteratively
+;; This is the core loop that implements the incremental selection strategy
+;;
+;; Parameters:
+;;   $itable - ITable for analysis
+;;   $remainingIndices - Feature indices not yet processed
+;;   $threshold - Relevance threshold
+;;   $maxTerms - Maximum interaction term size
+;;   $redundancyThreshold - Redundancy detection threshold
+;;   $currentTermSize - Current term size being processed (1, 2, 3, ...)
+;;   $selectedIndices - Indices selected in previous iterations
+;; Returns:
+;;   Final list of selected feature indices
+(= (incrementalIterator $itable $remainingIndices $threshold $maxTerms $redundancyThreshold $currentTermSize $selectedIndices)
+   (if (or (> $currentTermSize $maxTerms) (== (List.length $remainingIndices) 0))
+       $selectedIndices  ;; Termination condition
+       ;; Process current term size
+       (chain (generateCombinations (List.listToExpr $remainingIndices) $currentTermSize) $combinations
+       (chain (exprToList $combinations) $combinationList
+       (chain (filterRelevantCombinations $itable $combinationList $threshold) $relevantIndices
+        (chain (if (> $redundancyThreshold 0)
+                  (findRedundantFeatures $itable $relevantIndices (* $redundancyThreshold $threshold))
+                  Nil) $redundantIndices
+    ;   (chain  (() (println! (releavent redundant  $relevantIndices $redundantIndices ))) $debug1
+         (chain (List.difference $relevantIndices $redundantIndices) $nonRedundantIndices
+       (chain (List.union $selectedIndices $nonRedundantIndices) $newSelectedIndices
+       (chain (List.difference $remainingIndices $relevantIndices) $updatedRemainingIndices
+           ;; Recurse to next term size
+           (incrementalIterator $itable $updatedRemainingIndices $threshold $maxTerms 
+                                  $redundancyThreshold (+ $currentTermSize 1) $newSelectedIndices))))
+    ; )
+            )
+                                  
+          )
+        )
+                                  )))
+
+;; Main incremental feature selection algorithm
+;; Iteratively processes interaction terms of increasing size (1-way, 2-way, 3-way, etc.)
+;; For each term size: find relevant features, remove redundant ones, continue with remainder
+;;
+;; Parameters:
+;;   $itable - ITable containing features and target variable
+;;   $threshold - MI threshold for feature relevance (0.0 to 1.0)
+;;   $maxInteractionTerms - Maximum size of feature combinations to consider
+;;   $redundancyThreshold - Multiplier for redundancy detection (0 = no redundancy removal)
+;; Returns:
+;;   List of selected feature indices
+(= (incrementalFeatureSelection $itable $threshold $maxInteractionTerms $redundancyThreshold)
```

Comment:
Can you change the name to `incrementalSelection` to make it easier to map with the C++ version?